audio/decoders/raw.cpp

```diff
@@ -35,8 +35,8 @@ namespace Audio {
 // us to go with the macro approach. So far this is
 // the only template function that MSVC6 seemed to
 // compile incorrectly. Knock on wood.
-#define READ_ENDIAN_SAMPLE(is16Bit, isUnsigned, ptr, isLE) \
-	((is16Bit ? (isLE ? READ_LE_UINT16(ptr) : READ_BE_UINT16(ptr)) : (*ptr << 8)) ^ (isUnsigned ? 0x8000 : 0))
+#define READ_ENDIAN_SAMPLE(bytesPerSample, isUnsigned, ptr, isLE) \
+	((bytesPerSample == 1 ? (*ptr << 8) : (bytesPerSample == 2 ? (isLE ? READ_LE_UINT16(ptr) : READ_BE_UINT16(ptr)) : (int16)((isLE ? READ_LE_UINT24(ptr) : READ_BE_UINT24(ptr)) >> 8))) ^ (isUnsigned ? 0x8000 : 0))
```

Comment:
This is barely readable.
As the macro is used at only one place, why not putting it back where it's used and make it multiline with proper if structures?

Reply:
I can do that. I was not sure it was safe because of the comment above that macro:
> This used to be an inline template function, but buggy template function handling in MSVC6 forced  us to go with the macro approach. So far this is  the only template function that MSVC6 seemed to  compile incorrectly. Knock on wood.

That made me unsure whether moving that to where it is used, which is a template function, would work in MSVC6. But at the same time I don't know if anybody is still using that compiler.

Reply:
I have now pushed a commit that does that. If we think this is the way to go then the two commits can be squashed.

Reply:
I think the MSVC6 error was due to template function. As it is now, it's only some lines of code in the body of a regular function so I think it's safe.
And it's now really cleaner. :D Thanks.